### PR TITLE
386 header cart api fetch

### LINKED ISSUE #386: Header Cart Icon
**Cart Icon:**
 
image

 
The cart icon is driving from the API to the eCommerce system.  These all start with proxy-gateway.abcam.com and are looked up from the “shoppingBasketId” local storage session:
image
 
Using the “shoppingBasketId” local storage session you can then call the eCommerce function, for example, to get the cart add the basket ID and country:
 
https://proxy-gateway.abcam.com/ecommerce/rest/v1/basket/XXXXXXXX?country=XX
 
In my case, I have this:

https://proxy-gateway.abcam.com/ecommerce/rest/v1/basket/5ce2cd5c-0950-4bdf-a8fc-47eb602e37eb?country=GB
 
This then returns a list of items in my cart.  Based on the number of items, that is what displays on the cart icon:
image 
 
At this point, we should code in the option to pick up the number of items and show in the cart, but clicking on the icon should take you to the basket:
 
This again is based on “shoppingBasketId” local storage session and country selection:
 
https://www.abcam.com/en-us/shopping-basket/XXXXXXXXXXX?country=XX
 
As an example, mine is:
 
https://www.abcam.com/en-us/shopping-basket/5ce2cd5c-0950-4bdf-a8fc-47eb602e37eb?country=GB

### PR DESCRIPTION
**Pull request by kumarshoolam**
Fix #386 

Test URLs:
- Before: https://main--danaher-abcam--aemsites.aem.live/en-us/webinars/webinar-two
- After: https://386-headercart-api-fetch--danaher-abcam--aemsites.aem.live/en-us/webinars/webinar-two

### CODE REVIEW

File: eds/blocks/header/header.js [hunk 1]
```diff
@@ -419,4 +419,45 @@ export default async function decorate(block) {
...
+    const url = `https://proxy-gateway.abcam.com/ecommerce/rest/v1/basket/${shoppingBaskedId}?country=${selectedCountry}`;
+    fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
```
Comment (pardeepgera23) [inline eds/blocks/header/header.js:440]: log error in console

File: eds/blocks/header/header.js [hunk 2]
```diff
@@ -419,4 +419,45 @@ export default async function decorate(block) {
       dropdownLabel.querySelector('.user-icon-dd').style.transform = 'rotate(0deg)';
     }
   });
+
+  // Cart icon
+  const hostName = (!window.location.host.includes('localhost') && !window.location.host.includes('.hlx'))
+    ? window.location.host
+    : 'pp.abcam.com';
```
Comment (pardeepgera23) [inline eds/blocks/header/header.js:426]: no need to point to pp.abcam.com. Use window.location.host only.

**Review by pardeepgera23 — CHANGES_REQUESTED** [review]
added comments

File: eds/blocks/header/header.js [hunk 3]
```diff
@@ -419,4 +419,45 @@ export default async function decorate(block) {
...
+  const shoppingBaskedId = localStorage.getItem('shoppingBasketId');
+  const selectedCountry = (lastSelectedCountry !== null) ? lastSelectedCountry : 'US';
+
+  if (shoppingBaskedId !== null) {
+    const url = `https://proxy-gateway.abcam.com/ecommerce/rest/v1/basket/${shoppingBaskedId}?country=${selectedCountry}`;
+    fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
```
[the remark this answers is not in the document]
Comment (kumarshoolam) [reply eds/blocks/header/header.js]: Hello @pardeepgera23 
Addressed PR Review Comments & Made Changes Accordingly-386.
CC: @Meenu-GSPANN, @Rajesh-GSPANN

File: eds/blocks/header/header.js (hunk 1 above)
Comment (kumarshoolam) [reply eds/blocks/header/header.js:450]: Hello @pardeepgera23 
Addressed PR Review Comments & Made Changes Accordingly-386.
CC: @Meenu-GSPANN, @Rajesh-GSPANN

File: eds/blocks/header/header.js (hunk 2 above)
Comment (kumarshoolam) [reply eds/blocks/header/header.js]: Hello @pardeepgera23 
Addressed PR Review Comments & Made Changes Accordingly-386.
CC: @Meenu-GSPANN, @Rajesh-GSPANN

File: eds/blocks/header/header.js (hunk 1 above)
Comment (pardeepgera23) [reply eds/blocks/header/header.js:450]: Use // eslint-disable-next-line no-console

File: eds/blocks/header/header.js [hunk 4]
```diff
@@ -419,4 +428,51 @@ export default async function decorate(block) {
...
+  const productCount = parseInt(cartCountElement.textContent, 10) || 0;
+  if (productCount > 0) {
+    cartCountElement.classList.remove('hidden');
+  }
+  cartButton.addEventListener('click', () => {
+    if (productCount > 0) {
+      window.location.href = `https://${hostName}/en-us/shopping-basket/${shoppingBaskedId}?country=${selectedCountry}`;
+    } else {
+      // eslint-disable-next-line no-bitwise
+      alert(`You have ${productCount} products in your cart!`);
```
Comment (pardeepgera23) [inline eds/blocks/header/header.js:475]: Either use modal popup instead of alert or ignore the eslint for alert using // eslint-disable-next-line no-alert

**Review by pardeepgera23 — APPROVED** [review]

**Review by rgravitvl — APPROVED** [review]
Please address the redirect empty cart page, x-correlation-id and rest looks good to me

**Comment by kumarshoolam** [comment]
Hello @rgravitvl 

As we discussed, We have Addressed PR Comments regarding empty cart page redirection, x-correlation-id and worked accordingly.

CC: @Meenu-GSPANN , @pardeepgera23 , @Rajesh-GSPANN

**Pull request merged**